lib/ansible/runner.rb

```diff
@@ -237,6 +252,23 @@ def validate_params!(env_vars, extra_vars, tags, ansible_runner_method, playbook
         raise ArgumentError, errors.join("; ") if errors.any?
       end
 
+      def credentials_info(credentials, base_dir)
+        command_line = {}
+        env_vars     = {}
+        extra_vars   = {}
+        credentials.each do |id|
+          cred = Ansible::Runner::Credential.new(id, base_dir)
+
+          command_line.merge!(cred.command_line)
+          env_vars.merge!(cred.env_vars)
+          extra_vars.merge!(cred.extra_vars)
```

Comment:
So seems like there could be some clobbering of values and is not obvious to the user here.  Don't have a better solution in mind or think it should hold up the review, but just bringing it up as a topic for discussion for another time.

Reply:
Yeah, but I think it would be pretty rare. Plus I would thing the values from the credential should be the highest priority.

Reply:
But maybe we should check for conflicts and log?

Reply:
> Yeah, but I think it would be pretty rare.

I agree with this, for what it is worth.  Having multiple credential types being used all at once would be rare I think, but...

> Plus I would thing the values from the credential should be the highest priority.

The `credentials.each do |id|` above this implies that there could be multiple credentials of the same type, and when one overrides the other, it would be clear who would win out to the user (in practice, it is the last one in the list, of course, but that depends on how they get passed in).


> But maybe we should check for conflicts and log?

Yeah, I think this is probably the most pragmatic approach for now.  I would rather comment and warn about it when we see conflicts, but assume it won't usually be a problem for now instead of trying to over-engineer a solution for a problem that might not be there.